### PR TITLE
HUM-449 Audit shard md5sums

### PR DESCRIPTION
**Pull request by corystone**

### CODE REVIEW

File: objectserver/auditor.go [hunk 1]
```diff
@@ -62,6 +67,75 @@ type Auditor struct {
...
+	if err != nil || !finfo.Mode().IsRegular() {
+		return 0, fmt.Errorf("Object file isn't a normal file: %s", err)
+	}
+
+	bytes := int64(0)
+	if !skipMd5 {
+		file, err := os.Open(path)
+		if err != nil {
+			return bytes, fmt.Errorf("Error opening file: %s", err)
+		}
```
Comment (gholt) [inline objectserver/auditor.go]: Need a `defer file.Close()`?

File: objectserver/auditor.go [hunk 2]
```diff
@@ -62,6 +67,75 @@ type Auditor struct {
...
+		hash, ok := metadata["ETag"]
+		if !ok {
+			return bytes, fmt.Errorf("Metadata missing ETag: %s", string(metabytes))
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			return bytes, fmt.Errorf("Error opening file: %s", err)
+		}
```
Comment (gholt) [inline objectserver/auditor.go]: Need a `defer file.Close()`?

File: objectserver/auditor.go [hunk 3]
```diff
@@ -143,6 +217,109 @@ func auditHash(hashPath string, skipMd5 bool) (bytesProcessed int64, err error)
...
+	policyDir := filepath.Dir(dbpath)
+	path := filepath.Join(policyDir, "hec")
+	temppath := filepath.Join(filepath.Dir(policyDir), "tmp")
+	ringPartPower := bits.Len64(objRing.PartitionCount() - 1)
+	zapLogger, ok := a.logger.(*zap.Logger)
+	if !ok {
+		a.logger.Error("Logger type assertion failed")
+		zapLogger = zap.L()
+	}
+	db, err := NewIndexDB(dbpath, path, temppath, ringPartPower, 1, 32, zapLogger)
```
Comment (gholt) [inline objectserver/auditor.go]: This isn't going to work with the new changes where db_part_power and subdirs are policy config options.

File: objectserver/auditor.go [hunk 4]
```diff
@@ -143,6 +217,109 @@ func auditHash(hashPath string, skipMd5 bool) (bytesProcessed int64, err error)
...
+	if err != nil {
+		a.errors++
+		a.totalErrors++
+		a.logger.Error("Couldn't open indexdb", zap.String("dbpath", dbpath), zap.Error(err))
+		return
+	}
```
Comment (gholt) [inline objectserver/auditor.go]: Need a `defer db.Close()`?

**Comment by corystone** [comment]
Going to open a new PR with all the trimmings.

**Pull request closed without merging**